module/solutions/helper.js

```diff
@@ -2659,37 +2689,95 @@ module.exports = class SolutionsHelper {
 		userId,
 		pageSize,
 		pageNo,
-		search,
+		search = '',
 		filter,
 		surveyReportPage = '',
-		currentScopeOnly = false,
-		entityId = ''
+		currentScopeOnly = false
 	) {
 		return new Promise(async (resolve, reject) => {
 			try {
 				currentScopeOnly = UTILS.convertStringToBoolean(currentScopeOnly)
-				let assignedSolutions = await this.assignedUserSolutions(
-					solutionType,
-					userId,
-					search,
-					filter,
-					'',
-					'',
-					entityId
-				)
-				if (!assignedSolutions.success) {
+				if (currentScopeOnly && process.env.SUBMISSION_LEVEL == 'ENTITY') {
 					throw {
 						status: HTTP_STATUS_CODE.bad_request.status,
-						message: assignedSolutions.message,
+						message: CONSTANTS.apiResponses.CURRENT_SCOPE_ONLY_AND_ENTITYID_BOTH_CANNOT_BE_TRUE,
 					}
 				}
 				let totalCount = 0
 				let mergedData = []
 				let solutionIds = []
+				requestedData['filter'] = {}
+				let getTargetedSolution = true
+				let targetedSolutions = {
+					success: false,
+				}
+				let searchQuery
+				let aclVisibilityAllProjects
+				let aclVisibilitySpecificProjects
+				let aclVisibilityScopeProjects
 
-				if (assignedSolutions.success && assignedSolutions.data) {
-					totalCount = assignedSolutions.data.count
-					mergedData = assignedSolutions.data.data
+				if (filter && filter !== '') {
+					if (filter === CONSTANTS.common.CREATED_BY_ME) {
+						requestedData['filter']['isAPrivateProgram'] = {
+							$ne: false,
+						}
+					} else if (filter === CONSTANTS.common.ASSIGN_TO_ME) {
+						requestedData['filter']['isAPrivateProgram'] = false
+					} else if (filter === CONSTANTS.common.DISCOVERED_BY_ME) {
+						getTargetedSolution = false
+					}
+				}
+				// surveyReportPage = UTILS.convertStringToBoolean(surveyReportPage)
+
+				// if (filter === CONSTANTS.common.DISCOVERED_BY_ME) {
+				// 	getTargetedSolution = false
+				// }
+
+				let copiedRequestedData = _.cloneDeep(requestedData)
+
+				// Fetch all the targeted solutions
+				if (getTargetedSolution) {
+					targetedSolutions = await this.forUserRoleAndLocation(
+						copiedRequestedData,
+						solutionType,
+						'',
+						'',
+						CONSTANTS.common.DEFAULT_PAGE_SIZE,
```

Comment:
@Prajwal17Tunerlabs in diksha this passing of default page size caused some issue. can you check their lates code and see how they are managing it now? just check if there any difference, if not ignore

Reply:
DEFAULT_PAGE_SIZE & DEFAULT_PAGE_NO has been replaced with ml-core-projects code.

Reply:
no @VISHNUDAS-tunerlabs . I have now made the changes in ELEVATE-project as it is in ml-core-projects.
above mentioned variables are now removed in ELEVATE-Project is what i meant.